WordPress/wordpress-playground
# Remote.html: Use Playground CLI to produce minified WordPress build

Do not merge – explorations in progress

## Changes

Uses Playground CLI to produce a minified WordPress build, replacing the custom bash-based WordPres installation flow that relied on wget, unzip, sed, wp-cli etc.

This is to dogfood Playground CLI and put the boot protocol to a practical use. With this PR, CLI Playground produces a minified WordPress version that is later used to boot the web Playground. Eventually, the minification logic could be just a Blueprint instead of a Dockerfile. For now, though, I didn't explore that.

This PR also:

* in BasePHP, it ensures `/internal/shared/auto_prepend_file.php` is loaded even when evaling a code snippet and not a file – it turns out that wasn't the case before!
* Preloads the SQLite integration plugin from `/internal` in remote.html to avoid polluting the site
* Runs the WordPress installation wizard if needed

The WordPress installation wizard could likely be punted to another PR to keep the scope smaller.

## Remaining work

- [x] Test the ZIP export/import flow
- [x] Test the OPFS (browser storage) export/import flow
- [x] Test the NativeFS export/import flow
- [ ] Test the GitHub export/import flow

## Testing instructions

Rebuild the latest WordPress with

```shell
rm -rf packages/playground/wordpress-builds/public/wp-6.5
npx nx bundle-wordpress:major-and-beta playground-wordpress-builds
```

Then run `npm run dev` and confirm the local Playground loads without any issues.

Related: https://github.com/WordPress/wordpress-playground/issues/1398
